pkg/cloudprovider/providers/azure/azure_storage.go

```diff
@@ -19,23 +19,50 @@ package azure
 import (
 	"fmt"
 
+	"github.com/Azure/azure-sdk-for-go/arm/storage"
+	"github.com/Azure/go-autorest/autorest/to"
 	"github.com/golang/glog"
 )
 
+const (
+	defaultStorageAccountType  = string(storage.StandardLRS)
+	fileShareAccountNamePrefix = "file"
+)
+
 // CreateFileShare creates a file share, using a matching storage account
 func (az *Cloud) CreateFileShare(name, storageAccount, storageType, location string, requestGB int) (string, string, error) {
 	var err error
 	accounts := []accountWithLocation{}
-	if len(storageAccount) > 0 {
-		accounts = append(accounts, accountWithLocation{Name: storageAccount})
-	} else {
-		// find a storage account
-		accounts, err = az.getStorageAccounts()
-		if err != nil {
-			// TODO: create a storage account and container
-			return "", "", err
+	if len(storageAccount) == 0 {
+		// find a storage account that matches storageType
+		accounts, err = az.getStorageAccounts(storageType)
+		if err != nil || len(accounts) == 0 {
+			uniqueID := az.controllerCommon.resourceGroup + az.controllerCommon.location + az.controllerCommon.subscriptionID
+			storageAccount = getStorageAccountName(fileShareAccountNamePrefix, uniqueID)
+			if location == "" {
+				location = az.controllerCommon.location
+			}
+			if storageType == "" {
+				storageType = defaultStorageAccountType
+			}
+
+			glog.V(2).Infof("azureFile - no storage account found, begin to create a new storage account %s in resource group %s, location: %s, storageType: %s",
+				storageAccount, az.controllerCommon.resourceGroup, location, storageType)
+			cp := storage.AccountCreateParameters{
+				Sku:      &storage.Sku{Name: storage.SkuName(storageType)},
+				Tags:     &map[string]*string{"created-by": to.StringPtr("azure-file")},
+				Location: &location}
+			cancel := make(chan struct{})
+
+			_, errchan := az.controllerCommon.cloud.StorageAccountClient.Create(az.controllerCommon.resourceGroup, storageAccount, cp, cancel)
```

Comment:
Is it possible that the created account be used by 'blobDiskController' later? If it is, it's not perfect to name the account 'fileXXX'.

If we consider the two plugins (blob and file) are separated, the blob plugin should have some rule to probe storage account that it creates only? It's something we may take a look.

Reply:
@karataliu `fileXXX` will be only used for azure file. if `getStorageAccountName` func is ok, I will use this func to fix similar issue where storage account for azure disk is not found:
https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/providers/azure/azure_blobDiskController.go#L96